File: suc/kotlin/037.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.BottomAppBar
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.CenterAlignedTopAppBar
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.FloatingActionButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Shapes
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.geometry.Size
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.StrokeCap
import androidx.compose.ui.graphics.drawscope.Stroke
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat
import java.text.NumberFormat

private const val NAME = "448_BudgetPie_en"
private const val UI_TYPE = "Finance"
private const val STYLE_THEME = "Cold Gradient"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF4A5568)
        val secondary = Color(0xFF718096)
        val tertiary = Color(0xFFA0AEC0)
        val background = Color(0xFFF7FAFC)
        val surface = Color(0xFFFFFFFF)
        val surfaceVariant = Color(0xFFEDF2F7)
        val outline = Color(0xFFE2E8F0)
        val success = Color(0xFF48BB78)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFF56565)
        val onPrimary = Color(0xFFFFFFFF)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFF1A202C)
        val onBackground = Color(0xFF1A202C)
        val onSurface = Color(0xFF1A202C)

        val gradient1 = Brush.verticalGradient(listOf(Color(0xFF63B3ED), Color(0xFF4299E1)))
        val gradient2 = Brush.verticalGradient(listOf(Color(0xFF4FD1C5), Color(0xFF38B2AC)))
        val gradient3 = Brush.verticalGradient(listOf(Color(0xFF667EEA), Color(0xFF5A67D8)))
        val gradient4 = Brush.verticalGradient(listOf(Color(0xFFED64A6), Color(0xFFD53F8C)))
        val gradient5 = Brush.verticalGradient(listOf(Color(0xFFF6AD55), Color(0xFFED8936)))
    }

    object TypographyTokens {
        val display = TextStyle(fontSize = 36.sp, fontWeight = FontWeight.Bold, letterSpacing = (-0.5).sp)
        val headline = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 0.sp)
        val title = TextStyle(fontSize = 18.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.sp)
        val body = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Normal, letterSpacing = 0.sp)
        val label = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium, letterSpacing = 0.5.sp)
    }

    object Shapes {
        val small = RoundedCornerShape(4.dp)
        val medium = RoundedCornerShape(8.dp)
        val large = RoundedCornerShape(16.dp)
    }

    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 16.dp
        val lg = 24.dp
        val xl = 32.dp
        val xxl = 48.dp
    }
    
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(1.dp, 3.dp, 1.dp, 0.05f)
        val level2 = ShadowSpec(4.dp, 6.dp, 2.dp, 0.07f)
        val level3 = ShadowSpec(8.dp, 15.dp, 5.dp, 0.10f)
    }
}

private val AppColorScheme = lightColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    tertiary = AppTokens.Colors.tertiary,
    onTertiary = AppTokens.Colors.onTertiary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class BudgetItem(
    val name: String,
    val value: Double,
    val brush: Brush
)

val budgetData = listOf(
    BudgetItem("Housing", 1250.00, AppTokens.Colors.gradient1),
    BudgetItem("Food", 480.50, AppTokens.Colors.gradient2),
    BudgetItem("Transport", 210.75, AppTokens.Colors.gradient3),
    BudgetItem("Entertainment", 150.00, AppTokens.Colors.gradient4),
    BudgetItem("Utilities", 320.00, AppTokens.Colors.gradient5)
)

@Composable
fun PieChart(
    items: List<BudgetItem>,
    modifier: Modifier = Modifier,
    strokeWidth: Dp = 20.dp
) {
    val totalValue = items.sumOf { it.value }
    var startAngle = -90f

    Box(modifier = modifier, contentAlignment = Alignment.Center) {
        Canvas(modifier = Modifier.fillMaxSize()) {
            items.forEach { item ->
                val sweepAngle = (item.value / totalValue * 360).toFloat()
                drawArc(
                    brush = item.brush,
                    startAngle = startAngle,
                    sweepAngle = sweepAngle,
                    useCenter = false,
                    style = Stroke(width = strokeWidth.toPx(), cap = StrokeCap.Butt),
                    size = Size(size.width, size.height)
                )
                startAngle += sweepAngle
            }
        }
        Column(horizontalAlignment = Alignment.CenterHorizontally) {
             Text(
                text = "Total Spent",
                style = MaterialTheme.typography.labelMedium,
                color = AppTokens.Colors.secondary
            )
            Text(
                text = formatCurrency(totalValue),
                style = MaterialTheme.typography.headlineMedium,
                color = AppTokens.Colors.primary
            )
        }
    }
}

@Composable
fun BudgetItemRow(item: BudgetItem) {
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .padding(vertical = AppTokens.Spacing.md, horizontal = AppTokens.Spacing.md),
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.SpaceBetween
    ) {
        Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)) {
            Box(modifier = Modifier
                .size(20.dp)
                .background(item.brush, CircleShape))
            Text(text = item.name, style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.onSurface)
        }
        Text(text = formatCurrency(item.value), style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface)
    }
}

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    Scaffold(
        contentWindowInsets = WindowInsets(0),
        containerColor = AppTokens.Colors.background,
        topBar = {
            CenterAlignedTopAppBar(
                title = { Text("Monthly Budget", style = MaterialTheme.typography.headlineMedium, color = AppTokens.Colors.primary) }
            )
        },
        floatingActionButton = {
            FloatingActionButton(
                onClick = { /* Add new expense */ },
                containerColor = AppTokens.Colors.primary,
                contentColor = AppTokens.Colors.onPrimary,
                shape = CircleShape
            ) {
                 Text("+", style = MaterialTheme.typography.headlineMedium)
            }
        },
        bottomBar = {
            BottomAppBar(
                containerColor = AppTokens.Colors.surface,
                tonalElevation = AppTokens.ElevationMapping.level3.elevation,
            ) {
                Row(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(horizontal = AppTokens.Spacing.sm),
                    horizontalArrangement = Arrangement.SpaceAround,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    BottomBarItem(text = "Home", active = true)
                    BottomBarItem(text = "Reports", active = false)
                    Spacer(Modifier.width(AppTokens.Spacing.xl)) 
                    BottomBarItem(text = "Goals", active = false)
                    BottomBarItem(text = "Settings", active = false)
                }
            }
        }
    ) { padding ->
        LazyColumn(
            modifier = Modifier
                .fillMaxSize()
                .padding(padding)
                .padding(horizontal = AppTokens.Spacing.lg),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
        ) {
            item {
                PieChart(
                    items = budgetData,
                    modifier = Modifier.size(240.dp)
                )
                Spacer(modifier = Modifier.height(AppTokens.Spacing.lg))
            }
            items(budgetData) { item ->
                Card(
                    modifier = Modifier.fillMaxWidth(),
                    shape = AppTokens.Shapes.large,
                    colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                    elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level1.elevation)
                ) {
                    BudgetItemRow(item = item)
                }
            }
            item {
                Spacer(modifier = Modifier.height(AppTokens.Spacing.xxl))
            }
        }
    }
}

@Composable
fun BottomBarItem(text: String, active: Boolean) {
    Column(
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.xs)
    ) {
        val color = if (active) AppTokens.Colors.primary else AppTokens.Colors.tertiary
        Box(
            modifier = Modifier
                .size(24.dp)
                .background(color.copy(alpha = 0.2f), CircleShape)
        )
        Text(
            text = text,
            style = MaterialTheme.typography.labelMedium,
            color = color,
            textAlign = TextAlign.Center
        )
    }
}


private fun formatCurrency(value: Double): String {
    return NumberFormat.getCurrencyInstance(java.util.Locale.US).apply {
        maximumFractionDigits = 2
    }.format(value)
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())

        setContent {
            AppTheme {
                Surface(
                    modifier = Modifier.fillMaxSize(),
                    color = MaterialTheme.colorScheme.background
                ) {
                    RootScreen()
                }
            }
        }
    }
    
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, widthDp = 360, heightDp = 740)
@Composable
fun DefaultPreview() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/037.swift
import SwiftUI

// MARK: - Custom Hosting Controller to hide status bar
class CustomHostingController<Content: View>: UIHostingController<Content> {
    override var prefersStatusBarHidden: Bool {
        return true
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // 确保视图背景正确设置
        view.backgroundColor = UIColor(AppTokens.Colors.background)
    }
}

// MARK: - UIViewControllerRepresentable for Custom Hosting Controller
struct StatusBarHiddenView<Content: View>: UIViewControllerRepresentable {
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    func makeUIViewController(context: Context) -> CustomHostingController<Content> {
        return CustomHostingController(rootView: content)
    }
    
    func updateUIViewController(_ uiViewController: CustomHostingController<Content>, context: Context) {
        // 更新内容
    }
}

// MARK: - Main Entry Point
@main
struct BudgetPieApp: App {
    var body: some Scene {
        WindowGroup {
            StatusBarHiddenView {
                ContentView()
            }
        }
    }
}

// MARK: - App Tokens & Colors
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0x4A5568)
        static let secondary = Color(hex: 0x718096)
        static let tertiary = Color(hex: 0xA0AEC0)
        static let background = Color(hex: 0xF7FAFC)
        static let surface = Color(hex: 0xFFFFFFFF)
        static let surfaceVariant = Color(hex: 0xEDF2F7)
        static let outline = Color(hex: 0xE2E8F0)
        static let success = Color(hex: 0x48BB78)
        static let warning = Color(hex: 0xF59E0B)
        static let error = Color(hex: 0xF56565)
        static let onPrimary = Color(hex: 0xFFFFFFFF)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0x1A202C)
        static let onBackground = Color(hex: 0x1A202C)
        static let onSurface = Color(hex: 0x1A202C)
        
        // Gradients
        static let gradient1 = LinearGradient(colors: [Color(hex: 0x63B3ED), Color(hex: 0x4299E1)], startPoint: .top, endPoint: .bottom)
        static let gradient2 = LinearGradient(colors: [Color(hex: 0x4FD1C5), Color(hex: 0x38B2AC)], startPoint: .top, endPoint: .bottom)
        static let gradient3 = LinearGradient(colors: [Color(hex: 0x667EEA), Color(hex: 0x5A67D8)], startPoint: .top, endPoint: .bottom)
        static let gradient4 = LinearGradient(colors: [Color(hex: 0xED64A6), Color(hex: 0xD53F8C)], startPoint: .top, endPoint: .bottom)
        static let gradient5 = LinearGradient(colors: [Color(hex: 0xF6AD55), Color(hex: 0xED8936)], startPoint: .top, endPoint: .bottom)
    }
    
    struct Typography {
        static let display = Font.system(size: 36, weight: .bold)
        static let headline = Font.system(size: 24, weight: .semibold)
        static let title = Font.system(size: 18, weight: .medium)
        static let body = Font.system(size: 16, weight: .regular)
        static let label = Font.system(size: 14, weight: .medium)
    }
    
    struct Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 16
        static let lg: CGFloat = 24
        static let xl: CGFloat = 32
        static let xxl: CGFloat = 48
    }
    
    struct Shapes {
        static let small: CGFloat = 4
        static let medium: CGFloat = 8
        static let large: CGFloat = 16
    }
    
    struct Elevation {
        static let level1: CGFloat = 1
        static let level2: CGFloat = 4
        static let level3: CGFloat = 8
    }
}

// MARK: - Models
struct BudgetItem: Identifiable {
    let id = UUID()
    let name: String
    let value: Double
    let gradient: LinearGradient
}

// MARK: - Root View
struct ContentView: View {
    // Data
    let budgetData: [BudgetItem] = [
        BudgetItem(name: "Housing", value: 1250.00, gradient: AppTokens.Colors.gradient1),
        BudgetItem(name: "Food", value: 480.50, gradient: AppTokens.Colors.gradient2),
        BudgetItem(name: "Transport", value: 210.75, gradient: AppTokens.Colors.gradient3),
        BudgetItem(name: "Entertainment", value: 150.00, gradient: AppTokens.Colors.gradient4),
        BudgetItem(name: "Utilities", value: 320.00, gradient: AppTokens.Colors.gradient5)
    ]
    
    var body: some View {
        ZStack {
            // Background
            AppTokens.Colors.background.ignoresSafeArea()
            
            VStack(spacing: 0) {
                // Top Bar - 从屏幕顶部开始
                HStack {
                    Spacer()
                    Text("Monthly Budget")
                        .font(AppTokens.Typography.headline)
                        .foregroundColor(AppTokens.Colors.primary)
                        .padding(.vertical, AppTokens.Spacing.md)
                    Spacer()
                }
                .background(AppTokens.Colors.background)
                
                // Scrollable Content
                ScrollView(showsIndicators: false) {
                    VStack(spacing: AppTokens.Spacing.md) {
                        PieChart(items: budgetData, strokeWidth: 20)
                            .frame(width: 240, height: 240)
                            .padding(.vertical, AppTokens.Spacing.lg)
                        
                        // List Items
                        VStack(spacing: AppTokens.Spacing.md) {
                            ForEach(budgetData) { item in
                                CardView {
                                    BudgetItemRow(item: item)
                                }
                            }
                        }
                        
                        // Bottom Spacer for content padding
                        Spacer().frame(height: AppTokens.Spacing.xxl + 80)
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                }
            }
            
            // Bottom App Bar & FAB Overlay
            VStack {
                Spacer()
                
                ZStack(alignment: .top) {
                    // Bottom Bar Background
                    HStack(alignment: .center) {
                        BottomBarItem(text: "Home", active: true)
                        Spacer()
                        BottomBarItem(text: "Reports", active: false)
                        Spacer()
                        // Invisible placeholder for FAB space
                        Rectangle()
                            .fill(Color.clear)
                            .frame(width: 56)
                        Spacer()
                        BottomBarItem(text: "Goals", active: false)
                        Spacer()
                        BottomBarItem(text: "Settings", active: false)
                    }
                    .padding(.horizontal, AppTokens.Spacing.lg)
                    .frame(height: 80)
                    .background(AppTokens.Colors.surface)
                    .shadow(color: Color.black.opacity(0.1), radius: 15, x: 0, y: -5)
                    
                    // FAB
                    Button(action: {}) {
                        ZStack {
                            Circle()
                                .fill(AppTokens.Colors.primary)
                                .frame(width: 56, height: 56)
                                .shadow(radius: 4, y: 4)
                            
                            Image(systemName: "plus")
                                .font(.system(size: 24, weight: .semibold))
                                .foregroundColor(AppTokens.Colors.onPrimary)
                        }
                    }
                    .offset(y: -28)
                }
            }
        }
    }
}

// MARK: - Components

struct PieChart: View {
    let items: [BudgetItem]
    let strokeWidth: CGFloat
    
    // Helper struct to hold computed slice data
    struct SliceData: Identifiable {
        let id = UUID()
        let start: Angle
        let end: Angle
        let gradient: LinearGradient
    }
    
    // Compute slices based on items
    var slices: [SliceData] {
        let totalValue = items.map { $0.value }.reduce(0, +)
        var currentAngle = Angle(degrees: -90)
        var result: [SliceData] = []
        
        for item in items {
            let sweep = Angle(degrees: (item.value / totalValue) * 360)
            result.append(SliceData(start: currentAngle, end: currentAngle + sweep, gradient: item.gradient))
            currentAngle += sweep
        }
        return result
    }
    
    var totalValue: Double {
        items.map { $0.value }.reduce(0, +)
    }
    
    var body: some View {
        ZStack {
            ForEach(slices) { slice in
                PieSlice(startAngle: slice.start, endAngle: slice.end)
                    .stroke(slice.gradient, style: StrokeStyle(lineWidth: strokeWidth, lineCap: .butt))
            }
            
            VStack(spacing: 0) {
                Text("Total Spent")
                    .font(AppTokens.Typography.label)
                    .foregroundColor(AppTokens.Colors.secondary)
                Text(formatCurrency(totalValue))
                    .font(AppTokens.Typography.headline)
                    .foregroundColor(AppTokens.Colors.primary)
            }
        }
    }
}

struct PieSlice: Shape {
    let startAngle: Angle
    let endAngle: Angle
    
    func path(in rect: CGRect) -> Path {
        var path = Path()
        path.addArc(center: CGPoint(x: rect.midX, y: rect.midY),
                    radius: rect.width / 2,
                    startAngle: startAngle,
                    endAngle: endAngle,
                    clockwise: false)
        return path
    }
}

struct CardView<Content: View>: View {
    let content: Content
    
    init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }
    
    var body: some View {
        content
            .background(AppTokens.Colors.surface)
            .cornerRadius(AppTokens.Shapes.large)
            .shadow(color: Color.black.opacity(0.05), radius: 3, x: 0, y: 1)
    }
}

struct BudgetItemRow: View {
    let item: BudgetItem
    
    var body: some View {
        HStack(alignment: .center) {
            HStack(spacing: AppTokens.Spacing.md) {
                Circle()
                    .fill(item.gradient)
                    .frame(width: 20, height: 20)
                
                Text(item.name)
                    .font(AppTokens.Typography.body)
                    .foregroundColor(AppTokens.Colors.onSurface)
            }
            
            Spacer()
            
            Text(formatCurrency(item.value))
                .font(AppTokens.Typography.title)
                .foregroundColor(AppTokens.Colors.onSurface)
        }
        .padding(.vertical, AppTokens.Spacing.md)
        .padding(.horizontal, AppTokens.Spacing.md)
    }
}

struct BottomBarItem: View {
    let text: String
    let active: Bool
    
    var body: some View {
        VStack(spacing: AppTokens.Spacing.xs) {
            let color = active ? AppTokens.Colors.primary : AppTokens.Colors.tertiary
            
            Circle()
                .fill(color.opacity(0.2))
                .frame(width: 24, height: 24)
            
            Text(text)
                .font(AppTokens.Typography.label)
                .foregroundColor(color)
                .multilineTextAlignment(.center)
        }
    }
}

// MARK: - Helpers

func formatCurrency(_ value: Double) -> String {
    let formatter = NumberFormatter()
    formatter.numberStyle = .currency
    formatter.locale = Locale(identifier: "en_US")
    formatter.maximumFractionDigits = 2
    return formatter.string(from: NSNumber(value: value)) ?? "$0.00"
}

// MARK: - Color Hex Extension
extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xff) / 255,
            green: Double((hex >> 08) & 0xff) / 255,
            blue: Double((hex >> 00) & 0xff) / 255,
            opacity: alpha
        )
    }
}